### ISSUE TITLE
Linting unlinked files on change

### ISSUE THREAD

**Issue by KevinGhadyani-minted** [webpack-contrib/eslint-webpack-plugin#18]
- Operating System: Windows 10 v2004 64-bit
- Node Version: v12.18.4
- NPM Version: v6.14.6
- Yarn version: v1.22.5
- webpack Version: v4.44.2
- eslint-webpack-plugin Version: v2.1.0

### Expected Behavior

This only runs on changed files like `eslint-loader` did.

### Actual Behavior

This runs on all files in the context and matching the extensions of the files on your drive.

### Code

`webpack.config.js`
```js
const webpackConfig = {
  // Shouldn't matter what other configurations exist.
  // ...
  entry: 'src1/someFile1.js',
  plugins: [
    new ESLintWebpackPlugin({
      context: 'src1',
      extensions: ['.js'],
      fix: true,
    }),
    new ESLintWebpackPlugin({
      context: 'src2',
      extensions: ['.js'],
      fix: true,
    }),
  ],
  // ...
}
```

`eslintrc.js`
```js
const eslintConfig = {
  rules: {
    semi: 'error',
  }
}
```

`src1/someFile1.js`
```js
const someCode = Function.prototype;

someCode();
```

`src2/someFile2.js`
```js
const someOtherCode = Function.prototype;

someOtherCode();
```

### How Do We Reproduce?

1. Put those files in the correct folder structure.
2. Make a change in `someFile2.js` that should normally trigger an ESLint auto-fix (like removing a semi-colon) and save. This won't cause anything to trigger from Webpack because it's not being imported in an entrypoint.
3. Make a change in `someFile1.js` and `someFile2.js` will be ESLint auto-fixed.

This also occurs when starting Webpack as well as on-change. For instance, if I ran this to make a production build, it would still lint all changed files; even ones not included in an entrypoint.

When using the now-deprecated `eslint-loader`, this behavior only worked on files either included in the rule or part of an entrypoint. Sadly, `eslint-webpack-plugin` globally re-runs on all modified files when a change comes through on files that are imported from an entrypoint.

Either, it should only update files watched by an entrypoint _or_ it should run on-file-change for any valid files in its context. It should not do both.

**Comment by ricardogobbosouza** [comment]
Hi @KevinGhadyani-minted
Try [v2.2.0](https://github.com/webpack-contrib/eslint-webpack-plugin/releases/tag/v2.2.0)

**Comment by ricardogobbosouza** [comment]
If the problem persists using a v2.3.0 version, reopen this

**Comment by KevinGhadyani-minted** [comment]
Just to be clear, is the fix included in the `lintDirtyModulesOnly` prop or is it now part of standard operation?

**Comment by jsg2021** [comment]
This now only runs on the import graph generated by webpack. All that `lintDirtyModulesOnly` does is skip linting on the initial build. Only files altered that trigger a rebuild will be linted on subsequent runs.
